respx/mocks.py

```diff
@@ -297,14 +297,19 @@ def to_httpx_request(cls, **kwargs):
         Create a `HTTPX` request from transport request arg.
         """
         request = kwargs["request"]
+        method = (
+            request.method.decode("ascii").upper()
+            if isinstance(request.method, bytes)
```

Comment:
Should a `DeprecationWarning` be raised for bytes so that codebases can be updated and this support can eventually be dropped?

Reply:
Yes, I'd expect respx to follow the same argument conventions as httpx, and the project shouldn't try and maintain accepting bytes here if httpx has dropped support.

Reply:
Should the minimum HTTPX version also be bumped to match?

https://github.com/lundberg/respx/blob/366dd0bea824464e6ec9242a88f9b390a9dd74cb/setup.py#L44

Reply:
Suggested as https://github.com/ndhansen/respx/pull/1.

Reply:
> Should the minimum HTTPX version also be bumped to match?

I don’t think so, the library should still be compatible with older versions of httpx (they used to accept bytes for the `method` kwarg).

Reply:
I agree @mjpieters .. respx has always aligned with httpx and dropped support for old api's.

So, yes @hugovk, I think we should bump the required version of httpx .. respx is still compatible with older versions of httpx by using older versions of respx 😉